### ISSUE TITLE
Compiling issue - error[E0063]: missing field `gfa_output` in initializer of `GGCATConfig`

### ISSUE THREAD

**Issue by tmtktmtk** [jermp/fulgor#44]
The version 0f8190d7e51ef0faece835ec629811713ba224c0 cannot be compiled on linux fedora due to these following errors
```
   Compiling statrs v0.16.1
   Compiling bio v1.6.0
   Compiling genome-graph v9.0.0
   Compiling matchtigs v2.1.8
   Compiling ggcat_assembler v0.1.0 (/home/ktruong/Documents/code/original_fulgor/fulgor/external/ggcat/crates/assembler)
   Compiling ggcat-api v0.1.0 (/home/ktruong/Documents/code/original_fulgor/fulgor/external/ggcat/crates/api)
error[E0063]: missing field `gfa_output` in initializer of `GGCATConfig`
  --> crates/capi/src/lib.rs:19:42
   |
19 |     let instance = GGCATInstance::create(GGCATConfig {
   |                                          ^^^^^^^^^^^ missing `gfa_output`

error[E0061]: this method takes 11 arguments but 10 arguments were supplied
   --> crates/capi/src/lib.rs:93:10
    |
93  |           .build_graph(
    |  __________^^^^^^^^^^^-
94  | |             input_blocks,
95  | |             PathBuf::from(output_file),
96  | |             if color_names.len() > 0 {
...   |
118 | |             },
119 | |         )
    | |_________- argument #11 of type `bool` is missing
    |
note: method defined here
   --> /home/ktruong/Documents/code/original_fulgor/fulgor/external/ggcat/crates/api/src/lib.rs:199:12
    |
199 |     pub fn build_graph(
    |            ^^^^^^^^^^^
help: provide the argument
    |
93  ~         .build_graph(input_blocks, PathBuf::from(output_file), if color_names.len() > 0 {
94  +                 Some(color_names)
95  +             } else {
96  +                 None
97  +             }, kmer_length, threads_count, forward_only, if minimizer_length == usize::MAX {
98  +                 None
99  +             } else {
100 +                 Some(minimizer_length)
101 +             }, colors, min_multiplicity, match extra_elab {
102 +                 EXTRA_ELABORATION_STEP_NONE => ExtraElaboration::None,
103 +                 EXTRA_ELABORATION_STEP_UNITIG_LINKS => ExtraElaboration::UnitigLinks,
104 +                 EXTRA_ELABORATION_STEP_GREEDY_MATCHTIGS => ExtraElaboration::GreedyMatchtigs,
105 +                 EXTRA_ELABORATION_STEP_EULERTIGS => ExtraElaboration::Eulertigs,
106 +                 EXTRA_ELABORATION_STEP_PATHTIGS => ExtraElaboration::Pathtigs,
107 +                 _ => panic!("Invalid extra_elab value: {}", extra_elab),
108 +             }, /* bool */)
    |

Some errors have detailed explanations: E0061, E0063.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `ggcat-cpp-bindings` (lib) due to 2 previous errors
make[3]: *** [Makefile:15: lib/libggcat_cpp_bindings.a] Error 101
make[2]: *** [CMakeFiles/ggcat_cpp_api.dir/build.make:70: CMakeFiles/ggcat_cpp_api] Error 2
make[1]: *** [CMakeFiles/Makefile2:86: CMakeFiles/ggcat_cpp_api.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

```

**Comment by karel-brinda** [comment]
One comment – it seems to be potentially related to the CMake part of Fulgor. `cargo build` finishes without any issue if executed manually.

**Comment by Alessio-Campa** [comment]
The bug has been fixed by reverting the GGCAT update. If your last pulled commit is 0f8190d, executing the following commands should bring you to the correct commit.
```
git fetch
git reset HEAD~
git pull
git submodule update --init --recursive
```

Then run `make -j` in the build directory

**Comment by karel-brinda** [comment]
Actually, the previous commit was fixing another GGCAT issue on other platforms: https://github.com/jermp/fulgor/issues/41.

**Comment by karel-brinda** [comment]
Maybe moving towards the latest official release –  v1.1.0 – may fix the things?

**Comment by jermp** [comment]
ahah I totally forgot about the other GGCAT issue! So whatever we do, GGCAT refuses to compile :(

**Comment by luizirber** [comment]
It is an upstream error in GGCAT, they updated the C API but didn't test it before merging the change. I hit it in https://github.com/jermp/fulgor/pull/47#issue-2688776708 too...

**Comment by jermp** [comment]
Guys, I must admit I'm now a bit confused on how we should proceed to solve this issue. The PR @luizirber opened (Thank you so much!) should fix this?

**Comment by jermp** [comment]
Let me rephrase a bit: "The issue is to find a version of GGCAT that compiles/works everywhere" :-D

The release you mentioned is the one inducing the latest compilation error reported here in this issue...

**Comment by karel-brinda** [comment]
I think the version causing this issue was not the release (v1.1.0), but the latest head, no?
